src/service/search/grpc/mod.rs

```diff
@@ -210,3 +224,49 @@ pub async fn search(
 
     Ok(result)
 }
+
+pub fn handle_datafusion_error(err: DataFusionError) -> Error {
+    let err = err.to_string();
+    if err.contains("Schema error: No field named") {
+        let pos = err.find("Schema error: No field named").unwrap();
+        return match get_key_from_error(&err, pos) {
+            Some(key) => Error::ErrorCode(ErrorCodes::SearchFieldNotFound(key)),
+            None => Error::ErrorCode(ErrorCodes::SearchSQLExecuteError(err)),
+        };
+    }
+    if err.contains("parquet not found") {
+        return Error::ErrorCode(ErrorCodes::SearchParquetFileNotFound);
+    }
+    if err.contains("Invalid function ") {
+        let pos = err.find("Invalid function ").unwrap();
+        return match get_key_from_error(&err, pos) {
+            Some(key) => Error::ErrorCode(ErrorCodes::SearchFunctionNotDefined(key)),
+            None => Error::ErrorCode(ErrorCodes::SearchSQLExecuteError(err)),
+        };
+    }
+    if err.contains("Incompatible data types") {
```

Comment:
Please add a comment with an example of `err` value. Without an example, the parsing code below is indecipherable.